### PR TITLE
Remove TunableGroups from Storage classes and add TrialData tests

### PR DESCRIPTION
**Pull request by bpkroth**
Work split out from #633

- Reorgs the unit test fixtures for reuse by mlos_viz (future PR)
- Removes TunablesGroups from Storage classes. Currently unnecessary, belongs in Experiment, and causes inconsistencies with ExperimentData interactions.
- Adjusts the initialization code to match.

### CODE REVIEW

File: mlos_bench/mlos_bench/storage/base_trial_data.py [hunk 1]
```diff
@@ -46,38 +48,53 @@ def exp_id(self) -> str:
...
+    def _df_to_dict(self, dataframe: pandas.DataFrame) -> Dict[str, Optional[TunableValue]]:
+        """Utility function to convert certain key-value dataframe formats to a dict."""
+        if dataframe.columns.tolist() == ['metric', 'value']:
+            dataframe = dataframe.copy()
+            dataframe.rename(columns={'metric': 'parameter'}, inplace=True)
+        assert dataframe.columns.tolist() == ['parameter', 'value']
+        data = {}
+        for _, row in dataframe.iterrows():
+            assert isinstance(row['parameter'], str)
+            assert row['value'] is None or isinstance(row['value'], (str, int, float))
```
Comment (motus) [inline mlos_bench/mlos_bench/storage/base_trial_data.py:92]:
```python
from types import NoneType
```
and
```suggestion
            assert isinstance(row['value'], (NoneType, str, int, float))
```

File: mlos_bench/mlos_bench/tests/tunable_groups.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+import json5 as json
+
+from mlos_bench.config.schemas import ConfigSchema
+from mlos_bench.tunables.covariant_group import CovariantTunableGroup
+from mlos_bench.tunables.tunable_groups import TunableGroups
+
+# pylint: disable=redefined-outer-name
+
+TUNABLE_GROUPS_JSON = """
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/tunable_groups.py:21]: Don't we have it already in `conftest.py` on the same level? Maybe let's keep it there and reduce the duplication? (Also, I have changes coming to that JSON in my PR (adding weights, etc))

File: mlos_bench/mlos_bench/storage/base_trial_data.py (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/storage/base_trial_data.py:92]: We've had issues with NoneType support across python versions in the past, so I skipped this for now.

File: mlos_bench/mlos_bench/tests/tunable_groups.py (hunk 2 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups.py:21]: In #644 I'm reusing these definitions for mlos-viz tests.  We did something similar in `mlos_bench/mlos_bench/tests/services/remote/ssh/fixtures.py`.

I can separate those changes out again if you want, but it was easiest to apply those patches all at once here.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups.py:21]: Actually, might make sense to move those to a `fixtures.py` as well.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/tunable_groups.py]: And actually, in my original splitting of the work I'd forgotten to completely cleanup the old `conftest.py`.  Should be fixed now.  Please have a look and let me know what you think.  Thanks!

File: mlos_bench/mlos_bench/tests/environments/__init__.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 def check_env_success(env: Environment,
                       tunable_groups: TunableGroups,
-                      expected_results: Dict[str, Union[TunableValue]],
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/environments/__init__.py:22]: pylint complaint

**Review by motus — APPROVED** [review]

**Pull request merged**